pkg/api/v1/server_service.go

```diff
@@ -173,3 +179,54 @@ func (c *Client) ListVersionedAttributes(ctx context.Context, srvUUID uuid.UUID)
 
 	return *val, &r, nil
 }
+
+// CreateFirmware will attempt to create a firmware in Hollow and return the firmware UUID
+func (c *Client) CreateFirmware(ctx context.Context, firmware ComponentFirmwareVersion) (*uuid.UUID, *ServerResponse, error) {
+	resp, err := c.post(ctx, firmwaresEndpoint, firmware)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	u, err := uuid.Parse(resp.Slug)
+	if err != nil {
+		return nil, resp, nil
+	}
+
+	return &u, resp, nil
+}
+
+// DeleteFirmware will attempt to delete firmware and return an error on failure
+func (c *Client) DeleteFirmware(ctx context.Context, firmware ComponentFirmwareVersion) (*ServerResponse, error) {
+	return c.delete(ctx, fmt.Sprintf("%s/%s", firmwaresEndpoint, firmware.UUID))
+}
+
+// GetFirmware will return a firmware by its UUID
+func (c *Client) GetFirmware(ctx context.Context, fwUUID uuid.UUID) (*ComponentFirmwareVersion, *ServerResponse, error) {
+	path := fmt.Sprintf("%s/%s", firmwaresEndpoint, fwUUID)
```

Comment:
let's not use `Sprintf` and use [Golang's URL library instead](https://pkg.go.dev/net/url) that library will ensure that the fragments are URL-safe. We might be checking for safety elsewhere, but it's a good practice to verify.

Reply:
All the requests made by the client are already being `url.Parse()`ed in https://github.com/metal-toolbox/hollow-serverservice/blob/main/pkg/api/v1/requests.go#L16, do you think it's still necessary to `url.PathEncode()` here?